### ISSUE TITLE
0.48.7: redundantParens remove necessary parens around operators

### ISSUE THREAD

**Issue by jshier** [nicklockwood/SwiftFormat#970]
0.48.7 has started stripping parens that are actually necessary for operators to be used in certain situations. For example:

```swift
static func keyPath<Property: Comparable>(_ keyPath: KeyPath<Value, Property>,
                                          comparator: @escaping (Property, Property) -> Bool = (<)) -> Self
```

In 0.48.7, the `<` operator has its parens stripped leading to a build failure.

**Comment by nicklockwood** [comment]
@jshier thanks for reporting this, I'll get it fixed asap. Was this on an open source project? If you let me know which one I can add it to the regression test suite.

**Comment by jshier** [comment]
No, it's a private project. This part is roughly based on [Sundell's sorting article](https://www.swiftbysundell.com/articles/sorting-swift-collections/).

**Comment by nicklockwood** [comment]
@jshier fixed in 0.48.8

**Comment by jshier** [comment]
Thanks, works great!
